spec/models/revenue_estimation_spec.rb

```diff
@@ -19,6 +19,7 @@
     it { is_expected.to validate_presence_of(:revenue) }
     it { is_expected.to validate_presence_of(:date) }
     it { is_expected.to validate_numericality_of(:revenue) }
+    it { is_expected.to validate_numericality_of(:average_ticket).is_greater_than(0) }
```

Comment:
This is indeed a necessary and sufficient condition to prove both when it is invalid by a string and a non-zero input.
Soon you will realize the entire `context "invalid"` can be replaced by just this single line.
For now, please remove the invalid block since it is now redundant against the power of shoulda matcher. Shouldamatcher does all this for us under the hood.

Reply:
Perfect! Just remove the unseless specs.